xml/smb_native_os.xml

```diff
@@ -351,4 +351,196 @@
     <param pos="0" name="service.vendor" value="Wind River"/>
     <param pos="0" name="service.product" value="VxWorks CIFS"/>
   </fingerprint>
+
+  <fingerprint pattern="^OS/400 \D(\d+)\D(\d+)\D(\d+)">
+    <description>OS/400</description>
+    <example os.version="4" os.version.version="5" os.version.version.version="0">OS/400 V4R5M0</example>
+    <param pos="0" name="os.vendor" value="IBM"/>
+    <param pos="0" name="os.product" value="OS/400"/>
+    <param pos="1" name="os.version"/>
+    <param pos="2" name="os.version.version"/>
+    <param pos="3" name="os.version.version.version"/>
+  </fingerprint>
+
+  <!-- VisionFS -->
+
+  <fingerprint pattern="^(?:ax|i3|m8|mp|pa|pp|rs|sp)ai([\d]+4)">
```

Comment:
Is the `4` at the end expected?  Or are you trying to match anything with 4 digits at the end?